physical/optimizer/scenarios.go

```diff
@@ -39,6 +40,31 @@ var MergeRequalifiers = Scenario{
 	},
 }
 
+var RemoveEmptyMaps = Scenario{
+	Name:        "remove empty maps",
+	Description: "Removes maps that have no expressions and keep set to true",
+	CandidateMatcher: &MapMatcher{
+		Keep:        &AnyPrimitiveMatcher{Name: "keep"},
+		Expressions: &AnyNamedExpressionListMatcher{Name: "expressions"},
+		Source:      &AnyNodeMatcher{Name: "source"},
+	},
+
+	CandidateApprover: func(match *Match) bool {
+		expressions := match.NamedExpressionLists["expressions"]
+		keep := match.Primitives["keep"]
+		keepCast, ok := keep.(bool)
```

Comment:
I don't think you have to check the ok, just do
```
keepCast := keep.(bool)
```
if it's not a bool, then there's a serious programmatic error, and it shouldn't just return false.